### ISSUE TITLE
Plasma manager being marked dead because call to Create takes 60 seconds.

### ISSUE THREAD

**Issue by robertnishihara** [ray-project/ray#677]
While debugging #670, I came across the following problem.

The setup is the following.

- One head node (m4.16xlarge)
- 99 other nodes (c4.xlarge)
- Run the following workload on the head node

    ```python
    @ray.remote
    def f():
        return 1

    %time l = ray.get([f.remote() for _ in range(10000000)])
    ```

After several minutes, the plasma manager logs a fatal error for taking too long to send a heartbeat. Adding in some timing statements, the problem seems to be that the call

```c++
  Status s = conn->manager_state->plasma_conn->Create(
      object_id, data_size, NULL, metadata_size, &(buf->data));
```

in `process_data_request` in `plasma_manager.cc`, which creates an object in the object store (and which is a blocking call), takes **60 seconds** https://github.com/ray-project/ray/blob/96962cdee0c6803695505dfb3c3cdcc451ce7a5f/src/plasma/plasma_manager.cc#L911 (this is an absurd amount of time). Note that at this point in the script, that line has only been called about 700 times. All prior calls to that line took around 0ms, and all subsequent calls to that line took around 6ms. Note that the number 700 seems very small. This could be because the vast majority of tasks are getting executed locally.

**Comment by robertnishihara** [comment]
The problem is that the following. The store takes a while to get around to processing the manager's `Create` call because it is busy processing a get request from the driver for 10 million objects.

**Possible solution**
- Large **get** requests to the store should probably be broken down into smaller requests (e.g., perhaps 10000 IDs at a time).
- It shouldn't be necessary to address this problem, but large **wait** and **fetch** requests to the manager should probably be broken down as well.

Where should we break the request into smaller pieces? There are a couple of options.
1. In Python
2. In the plasma client
3. In the store and in the manager

**Comment by robertnishihara** [comment]
This problem seems to be fixed by the combination of #678 and starting 20 Redis shards.

With only 1 shard, I still see the following error printed by the Redis shard.

```
18733:M 17 Jun 01:58:16.386 # Client id=23 addr=172.31.20.0:50278 fd=14 name= age=2717 idle=2055 flags=N db=0 sub=3 psub=0 multi=-1 qbuf=0 qbuf-free=0 obl=16372 oll=8186 omem=134217732 events=rw cmd=subscribe scheduled to be closed ASAP for overcoming of output buffer limits.
```

**Comment by robertnishihara** [comment]
Closing because this is fixed as described above, but we need to give a better error message in the case where Redis gets overwhelmed.
